File: lib/libxl-3.8.1.0/examples/c/vs2012/custom/custom.c
#include <stdio.h>
#include <conio.h>
#include "libxl.h"

int main()
{
    BookHandle book = xlCreateBook();
    if(book) 
    {	
        int i, f[6];
        FormatHandle format[6];
        SheetHandle sheet;		

        f[0] = xlBookAddCustomNumFormat(book, L"0.0");
        f[1] = xlBookAddCustomNumFormat(book, L"0.00");
        f[2] = xlBookAddCustomNumFormat(book, L"0.000");
        f[3] = xlBookAddCustomNumFormat(book, L"0.0000");
        f[4] = xlBookAddCustomNumFormat(book, L"#,###.00 $");
        f[5] = xlBookAddCustomNumFormat(book, L"#,###.00 $[Black][<1000];#,###.00 $[Red][>=1000]");

        for(i = 0; i < 6; ++i) {
            format[i] = xlBookAddFormat(book, 0);
            xlFormatSetNumFormat(format[i], f[i]);
        }
	
        sheet = xlBookAddSheet(book, L"Custom formats", 0);
        if(sheet) 
        {
            xlSheetSetCol(sheet, 0, 0, 20, 0, 0);
            xlSheetWriteNum(sheet, 2, 0, 25.718, format[0]);
            xlSheetWriteNum(sheet, 3, 0, 25.718, format[1]);
            xlSheetWriteNum(sheet, 4, 0, 25.718, format[2]);
            xlSheetWriteNum(sheet, 5, 0, 25.718, format[3]);

            xlSheetWriteNum(sheet, 7, 0, 1800.5, format[4]);

            xlSheetWriteNum(sheet, 9, 0, 500, format[5]);
            xlSheetWriteNum(sheet, 10, 0, 1600, format[5]);            
        }

        if(xlBookSave(book, L"custom.xls")) printf("\nFile custom.xls has been created.\n");
        xlBookRelease(book);
    }

    printf("\nPress any key to exit...");
    _getch();

    return 0;
}

File: lib/libxl-3.8.1.0/examples/fortran/vs2008/custom/custom.f90
program custom

use libxl

type(BookHandle) book
type(SheetHandle) sheet, nullSheet
type(FormatHandle) f(6), nullFormat
integer c(6), i

book = xlCreateBook()

c(1) = xlBookAddCustomNumFormat(book, "0.0")
c(2) = xlBookAddCustomNumFormat(book, "0.00")
c(3) = xlBookAddCustomNumFormat(book, "0.000")
c(4) = xlBookAddCustomNumFormat(book, "0.0000")
c(5) = xlBookAddCustomNumFormat(book, "#,###.00 $")
c(6) = xlBookAddCustomNumFormat(book, "#,###.00 $[Black][<1000];#,###.00 $[Red][>=1000]")

do i = 1, 6  
  f(i) = xlBookAddFormat(book, nullFormat)
  call xlFormatSetNumFormat(f(i), c(i))
end do

sheet = xlBookAddSheet(book, "Custom formats", nullSheet)
    
i = xlSheetSetCol(sheet, 0, 0, 20, nullFormat, 0);
i = xlSheetWriteNum(sheet, 2, 0, 25.718, f(1));
i = xlSheetWriteNum(sheet, 3, 0, 25.718, f(2));
i = xlSheetWriteNum(sheet, 4, 0, 25.718, f(3));
i = xlSheetWriteNum(sheet, 5, 0, 25.718, f(4));

i = xlSheetWriteNum(sheet, 7, 0, 1800.5, f(5));

i = xlSheetWriteNum(sheet, 9, 0, 500, f(6));
i = xlSheetWriteNum(sheet, 10, 0, 1600, f(6));
    
i = xlBookSave(book, 'custom.xls')

call xlBookRelease(book)
    
end program custom